src/CalcManager/Ratpack/exp.cpp

```diff
@@ -342,11 +342,32 @@ void powratNumeratorDenominator(PRAT *px, PRAT y, uint32_t radix, int32_t precis
         DUPRAT(originalResult, pxPow);
         powratcomp(&originalResult, oneoveryDenom, radix, precision);
                 
+        // ##################################
+        // Find number of digits after decimal point
+        // and prepare multiplier for rounding
+        // ##################################
+        PRAT resultMultiplier = nullptr;
+        PRAT resultDigits = nullptr;
+        PNUMBER numResult = RatToNumber(pxPow, radix, precision);
+        if (-numResult->exp < precision)
+        {
+            DUPRAT(resultDigits, longtorat(-numResult->exp / 2));
```

Comment:
-numResult->exp / 2  truncates. Can you add a test where the number of digits after the decimal point is uneven?

Reply:
The 2 here only works for squares, not other powers. We've unlocked this PR for the purpose of being able to give feedback on the code itself, but let's keep discussing what the correct fix should be on the Issue.

And thanks, @uhliksk, for taking the time to dive into this issue!